runner/commander.go

```diff
@@ -33,13 +34,37 @@ func (c Commander) ID() string {
 	return c.Command
 }
 
+// CommandExists is a cross-platform function that returns true if a command exists on the host
+func CommandExists(command string) bool {
+	var lookup string
+
+	// Strip command of args; we're only testing the first token
+	command = strings.Fields(command)[0]
+
+	// Set appropriate lookup command based on OS
+	if runtime.GOOS == "windows" {
```

Comment:
I just realized that I can implement this via exec.LookPath(), which might be cleaner and faster (and just as cross-platform as this). Happy to hear your thoughts.

Reply:
Let's use `exec.LookPath()` in the implementation and make some adjustments to the error returns. CommandExists ought to return a `CommandNotFoundError` while wrapping the error from `exec.LookPath()`.  `ShellExecError` has a good example of how to implement the Unwrap() method and include the wrapped message in the `Error()` message.

Reply:
Could `command.autoDetectCommand` (in `command/run.go`) be extracted out to someplace shared and used for this? I think it does what is being looked for here, but it's just not exported currently. (`autoDetectCommand` could probably be given a more general name as well.)

Reply:
Thanks, both of you -- I've repurposed `command.autoDetectCommand` to fit Kit's description and moved to the `util` package (it just didn't seem right to put it into the command runner and then import that entire package in run.go). Let me know if there's a better place for it.